modules/extensions/process/publisher/process/themes/default/partials/view-asset.hbs

```diff
@@ -140,9 +140,147 @@
         });
     }
 
+    function isProcessNotAvailableInList(processName){
+        for(var i = 0 ; i < processNames.length ; i++){
+            if(processNames[i] == processName){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    function readUpdatedSubprocess(e){
+        var subprocessInfo = [];
+        var subprocessInput = $(e).parent().parent().closest("tr").find("input").val();
+
+        if(subprocessInput == ''){
+            alert("Field is empty");
+        }else if(isProcessNotAvailableInList(subprocessInput)){
+            alert("Please enter correct process name that exists.");
+        }else{
+            var subprocessPath, subprocessId;
+            for(var i = 0 ; i < processListObj.length ; i++){
+                if(processListObj[i].processname == subprocessInput.split("-")[0] &&
+                        processListObj[i].processversion == subprocessInput.split("-")[1]){
+                    subprocessPath = processListObj[i].path;
+                    subprocessId = processListObj[i].processid;
+                    break;
+                }
+            }
+            subprocessInfo.push({
+                name: subprocessInput.split("-")[0],
+                path: subprocessPath,
+                id: subprocessId
+            });
+            var subProcessDetails = {
+                'processName': $('#view-header').text(),
+                'processVersion' : $('#process-version').text(),
+                'subprocessList' : subprocessInfo
+            };
+
+            $.ajax({
+                url: 'https://localhost:9443/publisher/assets/process/apis/update_subprocess_list',
+                type: 'POST',
+                data: { 'subprocessDetails': JSON.stringify(subProcessDetails) },
+                success: function (response) {
+                    alert("Updated the subprocess details.");
+                },
+                error: function () {
+                    alert("error");
+                }
+            });
+        }
+    }
+
+    function readUpdatedSuccessor(e){
+        var successorInfo = [];
+        var successorInput = $(e).parent().parent().closest("tr").find("input").val();
+
+        if(successorInput == ''){
+            alert("Field is empty");
+        }else if(isProcessNotAvailableInList(successorInput)){
```

Comment:
Give as much as information with error/debug messages. E.g. mention which field is empty in the above message.